lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -233,14 +234,28 @@ class Chef::Knife::VsphereVmClone < Chef::Knife::BaseVsphereCommand
          description: 'Indicates whether to mark the new vm as a template',
          boolean: false
 
+  option :random_vmname,
+         long: '--random-vmname',
+         description: 'Creates a random VMNAME starts with vm-XXXXXXXX',
+         boolean: false
+
+  option :random_vmname_prefix,
+         long: '--random-vmname-prefix PREFIX',
+         description: 'Change the VMNAME prefix',
+         default: 'vm-'
+
   def run
     $stdout.sync = true
 
-    vmname = @name_args[0]
-    if vmname.nil?
+    unless using_supplied_hostname? ^ using_random_hostname?
       show_usage
-      fatal_exit('You must specify a virtual machine name')
+      fatal_exit('You must specify a virtual machine name OR use --random-vmname')
+    end
+
+    def random_hostname
```

Comment:
This method should probably go in the private section with the rest of them... Then I think we're done with this!